### ISSUE TITLE
Fail to use LinearTransformationScene by division by zero

### ISSUE THREAD

**Issue by yamauchih** [ManimCommunity/manim#780]
## Description of bug / unexpected behavior
When I use the (even empty) LinearTransformationScene, division by zero occurs.
Use ManimCommunity today's head. python3.7.5, poetry 1.1.4

The reason is 
manum/mobject/coordinate_systems.py:315, the 4th argument,

`   self.faded_line_ratio
`

is 0. Because of the manim/scene/vector_space_scene.py:5498 overwrites with 0 in the CONFIG of LinearTransformationScene.

Fix: set manim/scene/vector_space_scene.py:549 to

` "faded_line_ratio": 1,`

instead of 

` "faded_line_ratio": 0, # this causes division by zero`

generates an expected result. But I am not sure this is the correct way to fix.
Could anyone give me an advice?
Also I am new here and if this is not the right place to ask bugfix, please advise me.
Thanks a lot!
Hitoshi

## Expected behavior

Render a LinearTransformationScene (grid).

## How to reproduce the issue

<details><summary>Code for reproducing the problem</summary>

```python
from manim import *
class TestScene(LinearTransformationScene):
    def construct(self):
        self.wait(5)
```

</details>

## Additional media files

No image produced by an error.

<details><summary>Images/GIFs</summary>

</details>

## Logs
<details><summary>Terminal output</summary>

```
(manimce-s9WNd35N-py3.7) hitoshi@kubuntu1604[29]bash % python3 -m manim 00_linear.py  -p  -ql



Traceback (most recent call last):
  File "/home/hitoshi/data/gitdata/manim/community/manim/manim/__main__.py", line 84, in main
    scene.render()
  File "/home/hitoshi/data/gitdata/manim/community/manim/manim/scene/scene.py", line 88, in render
    self.setup()
  File "/home/hitoshi/data/gitdata/manim/community/manim/manim/scene/vector_space_scene.py", line 572, in setup
    self.plane = NumberPlane(**self.foreground_plane_kwargs)
  File "/home/hitoshi/data/gitdata/manim/community/manim/manim/mobject/coordinate_systems.py", line 277, in __init__
    self.init_background_lines()
  File "/home/hitoshi/data/gitdata/manim/community/manim/manim/mobject/coordinate_systems.py", line 290, in init_background_lines
    self.background_lines, self.faded_lines = self.get_lines()
  File "/home/hitoshi/data/gitdata/manim/community/manim/manim/mobject/coordinate_systems.py", line 319, in get_lines
    self.faded_line_ratio,
  File "/home/hitoshi/data/gitdata/manim/community/manim/manim/mobject/coordinate_systems.py", line 357, in get_lines_parallel_to_axis
    step = (1 / dense_freq) * freq
ZeroDivisionError: division by zero

```

</details>

## System specifications

<details><summary>System Details</summary>

- OS (Linux Kubuntu 18.04):
- RAM: 8G
- Python version (`Python 3.7.5):
- Installed modules (provide output from `pip list`):
```
Package                       Version
----------------------------- ----------
alabaster                     0.7.12
appdirs                       1.4.4
astroid                       2.4.2
attrs                         20.3.0
Babel                         2.9.0
black                         20.8b1
cairocffi                     1.2.0
certifi                       2020.11.8
cffi                          1.14.4
chardet                       3.0.4
click                         7.1.2
colorama                      0.4.4
colour                        0.1.5
commonmark                    0.9.1
cycler                        0.10.0
docutils                      0.16
guzzle-sphinx-theme           0.7.11
idna                          2.10
imagesize                     1.2.0
importlib-metadata            3.1.0
iniconfig                     1.1.1
isort                         5.6.4
Jinja2                        2.11.2
kiwisolver                    1.3.1
lazy-object-proxy             1.4.3
manimce                       0.1.0
MarkupSafe                    1.1.1
matplotlib                    3.3.3
mccabe                        0.6.1
mypy-extensions               0.4.3
numpy                         1.19.4
packaging                     20.4
pangocairocffi                0.4.0
pangocffi                     0.8.0
pathspec                      0.8.1
Pillow                        8.0.1
pip                           20.2.2
pluggy                        0.13.1
progressbar                   2.5
py                            1.9.0
pycairo                       1.20.0
pycparser                     2.20
pydub                         0.24.1
Pygments                      2.7.2
pylint                        2.6.0
pyparsing                     2.4.7
pytest                        6.1.2
python-dateutil               2.8.1
pytz                          2020.4
recommonmark                  0.6.0
regex                         2020.11.13
requests                      2.25.0
rich                          6.2.0
scipy                         1.5.4
setuptools                    49.6.0
six                           1.15.0
snowballstemmer               2.0.0
Sphinx                        3.3.1
sphinxcontrib-applehelp       1.0.2
sphinxcontrib-devhelp         1.0.2
sphinxcontrib-htmlhelp        1.0.3
sphinxcontrib-jsmath          1.0.1
sphinxcontrib-qthelp          1.0.3
sphinxcontrib-serializinghtml 1.1.4
toml                          0.10.2
tqdm                          4.54.0
typed-ast                     1.4.1
typing-extensions             3.7.4.3
urllib3                       1.26.2
wheel                         0.35.1
wrapt                         1.12.1
zipp                          3.4.0
```

</details>

<details><summary>LaTeX details</summary>

+ LaTeX distribution (e.g. TeX Live 2020):
+ Installed LaTeX packages:

No output
</details>

<details><summary>FFMPEG</summary>

Output of `ffmpeg -version`:

```
No output
```
</details>

**Comment by leotrs** [comment]
Hi @yamauchih and thanks for reporting this! 

Can you please post the output of your script (the one containing just the call to `self.wait(5)`) when you make the proposed change from 0 to 1?

Also, @cobordism, since we are getting rid of CONFIG and the problem here seems to be a faulty CONFIG key, could you please check whether we have the same problem on the remove-CONFIG branch?

**Comment by yamauchih** [comment]
Thanks for taking care of this. Here is the output.
1. When the CONFIG in class LinearTransformationScene,
"faded_line_ratio": 0,

```
(manimce-s9WNd35N-py3.7) hitoshi@[13]bash % python3 -m manim 00_linear.py -p -ql



Traceback (most recent call last):
  File "/home/hitoshi/data/gitdata/manim/community/manim/manim/__main__.py", line 84, in main
    scene.render()
  File "/home/hitoshi/data/gitdata/manim/community/manim/manim/scene/scene.py", line 88, in render
    self.setup()
  File "/home/hitoshi/data/gitdata/manim/community/manim/manim/scene/vector_space_scene.py", line 572, in setup
    self.plane = NumberPlane(**self.foreground_plane_kwargs)
  File "/home/hitoshi/data/gitdata/manim/community/manim/manim/mobject/coordinate_systems.py", line 277, in __init__
    self.init_background_lines()
  File "/home/hitoshi/data/gitdata/manim/community/manim/manim/mobject/coordinate_systems.py", line 290, in init_background_lines
    self.background_lines, self.faded_lines = self.get_lines()
  File "/home/hitoshi/data/gitdata/manim/community/manim/manim/mobject/coordinate_systems.py", line 319, in get_lines
    self.faded_line_ratio,
  File "/home/hitoshi/data/gitdata/manim/community/manim/manim/mobject/coordinate_systems.py", line 357, in get_lines_parallel_to_axis
    step = (1 / dense_freq) * freq
ZeroDivisionError: division by zero

```
2. When the CONFIG in class LinearTransformationScene,
"faded_line_ratio": 1,

```
(manimce-s9WNd35N-py3.7) hitoshi@[14]bash % python3 -m manim 00_linear.py -p -ql
[11/29/20 23:19:51] INFO     Animation 0 : Using cached data (hash :                                  caching.py:41
                             3070002022_2874490763_2253047108)                                                     
[11/29/20 23:19:53] INFO                                                                   scene_file_writer.py:562
                             File ready at /home/hitoshi/data/gitdata/ka_video/math/202011                         
                             18_bayes_manim/media/videos/00_linear/480p15/TestScene.mp4                            
                                                                                                                   
                    INFO     Rendered TestScene                                                         scene.py:96
                             Played 1 animations             

```

**Comment by leotrs** [comment]
Thanks! I meant if you could please show us the video or image that you get in the second instance (with faded_line_ratio=1) :slightly_smiling_face: Does it show what you expect it to show? Have you been able to otherwise use LinearTransformationScene with no changes other than setting faded_line_ratio to 1?

**Comment by yamauchih** [comment]
Yes, when faded_line_ratio=1, the result is expected (when faded_line_ratio=0, no result due to the error.) Here is the output image (this time: python3 -m manim 00_linear.py  -pi -ql, use -i option to get a gif image.)


![TestScene](https://user-images.githubusercontent.com/2727324/100603903-0e706780-3306-11eb-84e6-e7001d667d93.gif)

**Comment by leotrs** [comment]
Thank you! In that case, I think the best way to fix this is to add this modification to the ongoing PR that removes CONFIG. @cobordism

**Comment by cobordism** [comment]
I will add `faded_line_ratio=1` to the other PR

**Comment by yamauchih** [comment]
Hi @leotrs ,
Shall I need to do any action? I am not familiar with the process.
Thanks!

**Comment by cobordism** [comment]
@yamauchih 
You do not need to do anything.
Leotrs and I were in the middle of removing every single CONFIG dictionary from manim... and since the error you found was in a CONFIG dictionary, we decided to incorporate your fix into our work.
I have just added your fix here:
https://github.com/ManimCommunity/manim/pull/783/commits/90dae2afa83ab392f01fade924cf6ec1f7a8b714

As soon as PR 783 has been merged into manim master, the problem should be solved.
